collision_manager.py

```diff
@@ -256,22 +273,39 @@ def checkInternalCollisions(self):
         Returns:
             bool: whether or not there's a countable internal collision
         """
-        colliding, names = self.manager.is_collision_internal(trurn_names=True)
+        colliding, names = self.manager.in_collision_internal(return_names=True)
         if not colliding or not self.ignore_connected_links:
             return colliding
         # Here we assume *something* is in collision, we just need to know what exactly
         for name_tup in names:
             first = name_tup[0]
             second = name_tup[1]
+            if self.ignore_ee:
+                if first == self.arm.link_names[-1] or second == self.arm.link_names[-1]:
+                    continue
             link_index = self.arm.link_names.index(first)
             if link_index > 0 and self.arm.link_names[link_index - 1] == second:
                 continue
             elif link_index < self.num_links - 1 and self.arm.link_names[link_index + 1] == second:
                 continue
             else:
+                #print(names)
                 return True
         return False
 
+    def drawArmMeshes(self, ax):
+        """
+        Draw Arm Meshes (and update backup, displayable meshes)
+        Args:
+            ax: matplotlib axis object to draw to
+        """
+        joint_transforms = self.arm.getJointTransforms()
+        for i in range(self.num_links):
+            self.meshes[self.arm.link_names[i]].apply_transform(self.old_transforms[i].inv().gTM())
```

Comment:
not clear what's going on here, why apply old_transform then joint_transform?

Reply:
Descriptive comment added.